Add Docker support

pierlagreca opened the pull request:
This PR allows to build a docker image to run Faveo

pierlagreca committed:
Add docker support

giorio94 commented on Dockerfile:
```diff
@@ -0,0 +1,63 @@
+
+
+# Install necessary packages 
+RUN apt-get update -y && apt-get install -y apt-transport-https ca-certificates wget \
+    curl gnupg2 lsb-release sl mlocate dos2unix bash-completion openssl
```
Please, remove the packages which are not necessary, to avoid increasing the docker size. For instance, bash-completion and dos2unix should not be needed, as well as either curl or wget could be dropped. Also the other ones should be checked if necessary.

giorio94 commented on Dockerfile:
```diff
@@ -0,0 +1,63 @@
+RUN apt-get update -y && apt-get install -y apt-transport-https ca-certificates wget \
+    curl gnupg2 lsb-release sl mlocate dos2unix bash-completion openssl
+
+#RUN wget -O /etc/apt/trusted.gpg.d/php.gpg https://packages.sury.org/php/apt.gpg
```
Please, remove commented code

giorio94 commented on Dockerfile:
```diff
@@ -0,0 +1,63 @@
+
+# Add to crontab file
+
+RUN touch /etc/cron.d/faveo-cron
+
+RUN echo '* * * * * php /usr/share/nginx/artisan schedule:run > /dev/null 2>&1' >>/etc/cron.d/faveo-cron
+
+RUN chmod 0644 /etc/cron.d/faveo-cron
+
+RUN crontab /etc/cron.d/faveo-cron
+
+RUN sed -i "s/max_execution_time = .*/max_execution_time = 120/" /etc/php/7.1/fpm/php.ini
```
Please, group together related instructions to avoid creating too many layers.

giorio94 commented on default.conf:
```diff
@@ -0,0 +1,35 @@
+	# pass the PHP scripts to FastCGI server
+	#
+	location ~ \.php$ {
+		fastcgi_split_path_info ^(.+\.php)(/.+)$;
+		try_files $uri /index.php =404;
+		fastcgi_pass unix:/var/run/php/php7.1-fpm.sock;
+        fastcgi_send_timeout 600;
+        fastcgi_read_timeout 600;
+		fastcgi_index index.php;
+		fastcgi_param SCRIPT_FILENAME $document_root$fastcgi_script_name;
+		include fastcgi_params;
```
Ensure indentation is consistent.

giorio94 commented on docker-compose.yml:
```diff
@@ -0,0 +1,31 @@
+version: '3'
```
In production, we will not use docker-compose, so this might be dropped. Nonetheless, if useful for testing, feel free to ignore this comment.

giorio94 reviewed (commented):
Hi @pierlagreca, thanks for this PR.

Please, update the title and the description to clarify the purpose of this PR.
Additionally, I left some comments inline.

giorio94 commented on Dockerfile:
```diff
@@ -0,0 +1,54 @@
+
+# Install necessary packages 
+RUN apt-get update -y && apt-get install -y apt-transport-https ca-certificates wget \
+    gnupg2 lsb-release sl mlocate openssl
```
```suggestion
    gnupg2 lsb-release sl mlocate openssl && rm -rf /var/lib/apt/lists/*
```

Additionally, it seems to me `sl` and `mlocate` could be dropped. Should be tested though.

giorio94 reviewed (commented):
Overall, looks good to me.

Just a couple of minor comments. Then, please squash all commits and we can merge this.

pierlagreca commented:
I've just updated Dockerfile by removing `sl` and `mlocate`, I've removed `docker-compose.yml` and squashed all commits

giorio94 commented:
> I've just updated Dockerfile by removing `sl` and `mlocate`, I've removed `docker-compose.yml` and squashed all commits

Thanks. You should also fix the commit message (and possibly the author name) though.

pierlagreca commented:
> 
> 
> > I've just updated Dockerfile by removing `sl` and `mlocate`, I've removed `docker-compose.yml` and squashed all commits
> 
> Thanks. You should also fix the commit message (and possibly the author name) though.

I've done it now

giorio94 commented:
> > > I've just updated Dockerfile by removing `sl` and `mlocate`, I've removed `docker-compose.yml` and squashed all commits
> > 
> > 
> > Thanks. You should also fix the commit message (and possibly the author name) though.
> 
> I've done it now

The commit description still details the intermediate steps, and should be dropped

pierlagreca commented:
> 
> 
> > > > I've just updated Dockerfile by removing `sl` and `mlocate`, I've removed `docker-compose.yml` and squashed all commits
> > > 
> > > 
> > > Thanks. You should also fix the commit message (and possibly the author name) though.
> > 
> > 
> > I've done it now
> 
> The commit description still details the intermediate steps, and should be dropped

Sorry, I've dropped now the description, I hope everything is ok

giorio94 commented:
> > > > > I've just updated Dockerfile by removing `sl` and `mlocate`, I've removed `docker-compose.yml` and squashed all commits
> > > > 
> > > > 
> > > > Thanks. You should also fix the commit message (and possibly the author name) though.
> > > 
> > > 
> > > I've done it now
> > 
> > 
> > The commit description still details the intermediate steps, and should be dropped
> 
> Sorry, I've dropped now the description, I hope everything is ok

Seems fine :+1:
